### PR TITLE
Add special transaction(mnhf) for ehf

### PR DESCRIPTION
**Pull request by pravblockc**
The goal of this PR is to create a special transaction to support EnhancedHardFork(EHF) which is DIP23.
This would address the issue https://github.com/dashpay/dash/issues/4533

### CODE REVIEW

File: src/evo/mnhftx.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    MNHFTxPayload mnhfTx;
+    if (!GetTxPayload(tx, mnhfTx)) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-mnhf-payload");
+    }
+
+    if (mnhfTx.nVersion == 0 || mnhfTx.nVersion > MNHFTxPayload::CURRENT_VERSION) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-mnhf-version");
+    }
+
+    const CBlockIndex* pindexQuorum = LookupBlockIndex(mnhfTx.signal.quorumHash);
```
Comment (PastaPastaPasta) [inline src/evo/mnhftx.cpp:49]: We are potentially not locking cs_main here

File: src/evo/mnhftx.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#include <consensus/validation.h>
+
+#include <chain.h>
+#include <chainparams.h>
+#include <logging.h>
+#include <validation.h>
+
+extern const std::string CBLSIG_REQUESTID_PREFIX = "clsig";
+
+bool MNHFTx::Verify(const CBlockIndex* pQuorumIndex, bool checkSigs) const
```
Comment (PastaPastaPasta) [inline src/evo/mnhftx.cpp:18]: checkSigs is unused here, I think we can just drop it, I think we should always validate sigs here?

File: src/evo/mnhftx.cpp (hunk 1 above)
Comment (pravblockc) [reply src/evo/mnhftx.cpp:49]: added exclusive lock as per recommendation

**Comment by pravblockc** [comment]
I messed up rebas, so going to raise new PR

**Pull request closed without merging**